lib/midiHandler.py

```diff
@@ -0,0 +1,295 @@
+__author__ = 'PMarchant'
+
+import os
+import json
+import math
+import pprint
+import logging, logging.handlers, logging.config
+from tracemalloc import BaseFilter
+from music21 import *
+from talkingscores.settings import BASE_DIR, MEDIA_ROOT, STATIC_ROOT, STATIC_URL
+
+logger = logging.getLogger("TSScore")
+
+class MidiHandler:
+    def __init__(self, get, folder, filename):
+        self.queryString = get
+        self.folder = folder
+        self.filename = filename.replace(".mid", "")
+
+    #get list of selected / unselected instruments from binary of number.  Leftmost value is always 1
+    def get_selected_instruments(self):
+        bsi = int(self.queryString.get("bsi"))
+        self.selected_instruments = []
+        while (bsi>1):
+            print ("bsi = " + str(bsi))
+            if (bsi&1==True):
+                self.selected_instruments.append(True)
+            else:
+                self.selected_instruments.append(False)
+            bsi=bsi>>1
+        self.selected_instruments.reverse()
+        print(self.selected_instruments)
+
+        self.all_selected_parts = []
+        self.all_unselected_parts = []
+        self.selected_instruement_parts = {} #key = instrument, value = [parts]
+        
+        instrument_index=-1
+        prev_instrument=""
+        for part_index, part in enumerate(self.score.flat.getInstruments()):
+            print ("part_index = " + str(part_index) )
+            print (part)
+            if part.partId!=prev_instrument:
```

Comment:
I'd highly recommend installing a "linter" plugin such as autopep8 to let it handle formatting your code to a common standard.

Reply:
I have now installed autopep8.  

I wanted it to fix the formatting as I type - in settings.json
"editor.formatOnType": true,
But that didn't seem work.  

If I right click and choose format docuent - that works.
Yup - I made that mistake in another project - I wanted to format the few lines I'd written - but it did the entire file and my diff was incomprehensible!

However - if I right click and do Format Selection - I get a message saying "Configure default formatter.  Extension 'autopep8' is configured as formatter but it cannot format 'python -files"

Reply:
Fixed in 74276ac
I couldn't get it to work on just selected text - but the whole file works fine.
It does make a big difference to how the code looks!
I increased the maximum line length - because it was putting comments after the line they refer to - whereas I like them on the line or on the line before.